src/main/java/cz/student/simon8491/UserDao.java

```diff
@@ -0,0 +1,115 @@
+package cz.student.simon8491;
+
+import org.mindrot.jbcrypt.BCrypt;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class UserDao {
+    private static final String CREATE_USER_QUERY = """
+            INSERT INTO users (username, email, password) VALUES (?, ?, ?)""";
+    private static final String READ_USER_QUERY = """
+            SELECT * FROM users WHERE id = ?""";
+    private static final String UPDATE_USER_QUERY = """
+            UPDATE users SET username = ?, email = ?, password = ? WHERE id = ?""";
+    private static final String DELETE_USER_QUERY = """
+            DELETE FROM users WHERE id = ?""";
+    private static final String FIND_ALL_USERS_QUERY = """
+            SELECT * FROM users""";
+
+    public void createUser(User user) {
+        try (Connection connection = DBUtil.getConnection()) {
+            try (var stmtCreateUser = connection.prepareStatement
+                    (CREATE_USER_QUERY, PreparedStatement.RETURN_GENERATED_KEYS)) {
+                stmtCreateUser.setString(1, user.getUserName());
+                stmtCreateUser.setString(2, user.getEmail());
+                stmtCreateUser.setString(3, hashPassword(user.getPassword()));
+                stmtCreateUser.executeUpdate();
+                try (var rsCreateUser = stmtCreateUser.getGeneratedKeys()) {
+                    if (rsCreateUser.next()) {
+                        user.setId(rsCreateUser.getInt(1));
+                        System.out.println("Created user with ID: " + user.getId());
+                    }
+                }
+            }
+        } catch (SQLException e) {
+            System.err.println("Database connection error\n" + e.getMessage());
```

Comment:
IMRPOVE: zlepsi logovani vyvjimky i s stacktracem je to dobre pak pro hledani chyby. Prosim upravit i u dalsich metod. Navic text vyjimy neni uplne nejlepsi "Database connection error", muze byt pripojeni do db ok, ale problem muze byt neexistujici tabulky nebo nedostatecne prava. Proto bych radsi zvolil "Problem with create user in database"

```suggestion
System.err.println("Problem with create user in database: " + e.getMessage());
e.printStackTrace(System.err);
```